arangod/Transaction/Methods.cpp

```diff
@@ -2506,6 +2507,9 @@ Future<OperationResult> transaction::Methods::truncateLocal(std::string const& c
           }
         }
       }
+      if (findRefusal(responses)) {
+        return futures::makeFuture(OperationResult(TRI_ERROR_CLUSTER_SHARD_LEADER_RESIGNED));
+      }
```

Comment:
why was this code moved aroung?

Reply:
Ah! This is indeed part of the fix! Otherwise the follower is not dropped and thus not resynced!

Reply:
Indeed so.